src/pages/settings/Profile/CustomStatus/StatusClearAfterPage.js

```diff
@@ -0,0 +1,236 @@
+        customDateTemporary: PropTypes.string,
+    }),
+};
+
+/**
+ * @param {string} data - one of the values from CONST.CUSTOM_STATUS_TYPES or data in format YYYY-MM-DD HH:mm
+ * @returns {string}
+ */
+function getInitialDateBasedFromStatusType(data) {
+    switch (data) {
+        case DateUtils.getEndOfToday():
+            return CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY;
+        case CONST.CUSTOM_STATUS_TYPES.NEVER:
+        case '':
+            return CONST.CUSTOM_STATUS_TYPES.NEVER;
+        case false:
+            return CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY;
+        default:
+            return CONST.CUSTOM_STATUS_TYPES.CUSTOM;
+    }
+}
+
+const useValidateCustomDate = (data) => {
+    const localize = useLocalize();
+    const [customDateError, setCustomDateError] = useState('');
+    const [customTimeError, setCustomTimeError] = useState('');
+
+    const validate = () => {
+        const inputData = moment(data, 'YYYY-MM-DD HH:mm:ss');
+        const currentDate = moment().startOf('day');
+        const currentDateTimePlus2 = moment().add(1, 'minutes');
+
+        // Date validation
+        if (inputData.isBefore(currentDate)) {
+            setCustomDateError(localize.translate('common.error.invalidDateShouldBeFuture'));
+        } else {
+            setCustomDateError('');
+        }
+
+        // Time validation
+        if (inputData.isBefore(currentDateTimePlus2)) {
+            setCustomTimeError(localize.translate('common.error.invalidTimeShouldBeFuture'));
+        } else {
+            setCustomTimeError('');
+        }
+    };
+
+    useEffect(() => {
+        validate();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [data]);
+
+    const triggerValidation = () => {
+        validate();
+    };
+
+    return {customDateError, customTimeError, triggerValidation};
+};
+
+function StatusClearAfterPage({currentUserPersonalDetails, customStatus}) {
+    const localize = useLocalize();
+    const clearAfter = lodashGet(currentUserPersonalDetails, 'status.clearAfter', '');
+    const draftClearAfter = lodashGet(customStatus, 'clearAfter', '');
+    const customDateTemporary = lodashGet(customStatus, 'customDateTemporary', '');
+    const [draftPeriod, setDraftPeriod] = useState(getInitialDateBasedFromStatusType(clearAfter || draftClearAfter));
+    const localesToThemes = useMemo(
+        () =>
+            _.map(CONST.CUSTOM_STATUS_TYPES, (value, key) => ({
+                value,
+                text: localize.translate(`statusPage.timePeriods.${value}`),
+                keyForList: key,
+                isSelected: draftPeriod === value,
+            })),
+        [draftPeriod, localize],
+    );
+
+    const {customDateError, customTimeError, triggerValidation} = useValidateCustomDate(customDateTemporary);
+
+    const {brickDateIndicator, brickTimeIndicator} = useMemo(
+        () => ({
+            brickDateIndicator: customDateError ? CONST.BRICK_INDICATOR.ERROR : null,
```

Comment:
Might be useful to specific that it's `redBrick`.

Reply:
👍🏻